modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/SpacesSectionDisplayContext.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.site.cms.site.initializer.internal.display.context;
+
+import com.liferay.headless.asset.library.dto.v1_0.AssetLibrary;
+import com.liferay.headless.asset.library.resource.v1_0.AssetLibraryResource;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Roberto Díaz
+ */
+public class SpacesSectionDisplayContext {
+
+	public SpacesSectionDisplayContext(
+		HttpServletRequest httpServletRequest,
+		AssetLibraryResource.Factory assetLibraryResourceFactory) {
+
+		_assetLibraryResourceFactory = assetLibraryResourceFactory;
+
+		_themeDisplay = (ThemeDisplay)httpServletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+	}
+
+	public Page<AssetLibrary> getPage() throws Exception {
+		AssetLibraryResource.Builder builder =
+			_assetLibraryResourceFactory.create();
+
+		AssetLibraryResource assetLibraryResource = builder.user(
+			_themeDisplay.getUser()
+		).build();
+
+		Page<AssetLibrary> assetLibrariesPinnedByMePage =
+			assetLibraryResource.getAssetLibrariesPinnedByMePage(
+				Pagination.of(1, 5));
+
+		Page<AssetLibrary> assetLibrariesPage =
+			assetLibraryResource.getAssetLibrariesPage(
+				null, null, null, Pagination.of(1, 5), null);
+
+		return Page.of(
+			assetLibrariesPage.getActions(),
+			_getItems(assetLibrariesPinnedByMePage, assetLibrariesPage),
+			Pagination.of(1, 5), assetLibrariesPage.getTotalCount());
+	}
+
+	private Collection<AssetLibrary> _getItems(
+		Page<AssetLibrary> assetLibraryPinsPage,
+		Page<AssetLibrary> assetLibrariesPage) {
+
+		List<AssetLibrary> assetLibrariesPageItems = new ArrayList<>(
+			assetLibrariesPage.getItems());
+
+		if (assetLibrariesPageItems.isEmpty()) {
+			Collections.emptyList();
+		}
+
+		List<AssetLibrary> items = new ArrayList<>(
+			assetLibraryPinsPage.getItems());
+
+		if (items.size() == 5) {
+			return items;
+		}
+
+		List<Long> assetLibraryIdsList = ListUtil.toList(
+			items, item -> item.getId());
```

Comment:
ListUtil.toList already copies the source list, so there's no need to create the initial copy.